components/ILIAS/Authentication/classes/Password/class.ilLocalPasswordSettingsGUI.php

```diff
@@ -0,0 +1,249 @@
+        $section = $form->getInputs()['password'];
+        /** @var Password $cp
+         * @var Password $np
+         */
+        $cp = $section->getInputs()['current_password'];
+        $np = $section->getInputs()['new_password'];
+        $errors = ['current_password' => [], 'new_password' => []];
+
+        $entered_current_password = $cp->getValue();
+        $entered_new_password = $np->getValue();
+
+        if (!$form->getError()) {
+            $data = $form->getData();
+            $error = false;
+
+            if ((int) $this->user->getAuthMode(true) === ilAuthUtils::AUTH_LOCAL) {
+                if (!ilLocalPasswordManager::getInstance()->verifyPassword(
+                    $this->user,
+                    $entered_current_password
+                )) {
+                    $error = true;
+                    $errors['current_password'][] = $this->lng->txt('passwd_wrong');
+                }
+            }
+
+            if (!ilSecuritySettingsChecker::isPassword($entered_new_password, $custom_error)) {
+                $error = true;
+                if ($custom_error !== '') {
+                    $errors['new_password'][] = $custom_error;
+                } else {
+                    $errors['new_password'][] = $this->lng->txt('passwd_invalid');
+                }
+            }
+            $error_lng_var = '';
+            if (!ilSecuritySettingsChecker::isPasswordValidForUserContext(
+                $entered_new_password,
+                $this->user,
+                $error_lng_var
+            )) {
+                $this->tpl->setOnScreenMessage('failure', $this->lng->txt('form_input_not_valid'));
+                $errors['new_password'][] = $this->lng->txt($error_lng_var);
+                $error = true;
+            }
+            if (
+                $entered_current_password === $entered_new_password &&
+                ($this->user->isPasswordExpired() || $this->user->isPasswordChangeDemanded())
+            ) {
+                $error = true;
+                $errors['new_password'][] = $this->lng->txt('new_pass_equals_old_pass');
+            }
+
+            if (!$error) {
+                $this->user->resetPassword($entered_new_password, $entered_new_password);
+                if ($entered_current_password !== $entered_new_password) {
+                    $this->user->setLastPasswordChangeToNow();
+                    $this->user->setPasswordPolicyResetStatus(false);
+                    $this->user->update();
+                }
+
+                if (ilSession::get('orig_request_target')) {
+                    $this->tpl->setOnScreenMessage('success', $this->lng->txt('saved_successfully'), true);
+                    $target = ilSession::get('orig_request_target');
+                    ilSession::set('orig_request_target', '');
+                    ilUtil::redirect($target);
+                } else {
+                    $this->tpl->setOnScreenMessage('success', $this->lng->txt('saved_successfully'));
```

Comment:
```suggestion
                    $this->tpl->setOnScreenMessage($this->tpl::MESSAGE_TYPE_SUCCESS, $this->lng->txt('saved_successfully'));
```

Reply:
Or even better: Can this re replaced with a `UI/Message > Success` element and (because no HTTP redirect is done) passed to `showPassword` (where it should be rendered above the form)?